edx_user_state_client/tests.py

```diff
@@ -0,0 +1,349 @@
+        self.set(0, 0, {'a': 'b'})
+        self.assertEquals(self.get(0, 0, ['a', 'b']), {'a': 'b'})
+
+    def test_set_two_users(self):
+        self.set(0, 0, {'a': 'b'})
+        self.set(1, 0, {'b': 'c'})
+        self.assertEquals(self.get(0, 0), {'a': 'b'})
+        self.assertEquals(self.get(1, 0), {'b': 'c'})
+
+    def test_set_two_blocks(self):
+        self.set(0, 0, {'a': 'b'})
+        self.set(0, 1, {'b': 'c'})
+        self.assertEquals(self.get(0, 0), {'a': 'b'})
+        self.assertEquals(self.get(0, 1), {'b': 'c'})
+
+    def test_set_many(self):
+        self.set_many(0, {0: {'a': 'b'}, 1: {'b': 'c'}})
+        self.assertEquals(self.get(0, 0), {'a': 'b'})
+        self.assertEquals(self.get(0, 1), {'b': 'c'})
+
+    def test_get_many(self):
+        self.set_many(0, {0: {'a': 'b'}, 1: {'b': 'c'}})
+        self.assertItemsEqual(
+            self.get_many(0, [0, 1]),
+            [
+                (self._block(0), {'a': 'b'}),
+                (self._block(1), {'b': 'c'})
+            ]
+        )
+
+    def test_delete(self):
+        with self.assertRaises(self.client.DoesNotExist):
+            self.get(0, 0)
+
+        self.set(0, 0, {'a': 'b'})
+        self.assertEqual(self.get(0, 0), {'a': 'b'})
+
+        self.delete(0, 0)
+        with self.assertRaises(self.client.DoesNotExist):
+            self.get(0, 0)
+
+    def test_delete_partial(self):
+        with self.assertRaises(self.client.DoesNotExist):
+            self.get(0, 0)
+
+        self.set(0, 0, {'a': 'b', 'b': 'c'})
+        self.assertEqual(self.get(0, 0), {'a': 'b', 'b': 'c'})
+
+        self.delete(0, 0, ['a'])
+        self.assertEqual(self.get(0, 0), {'b': 'c'})
+
+    def test_delete_last_field(self):
+        with self.assertRaises(self.client.DoesNotExist):
+            self.get(0, 0)
+
+        self.set(0, 0, {'a': 'b'})
+        self.assertEqual(self.get(0, 0), {'a': 'b'})
+
+        self.delete(0, 0, ['a'])
+        with self.assertRaises(self.client.DoesNotExist):
+            self.get(0, 0)
+
+    def test_delete_many(self):
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [])
+
+        self.set_many(0, {
+            0: {'a': 'b'},
+            1: {'b': 'c'},
+        })
+
+        self.delete_many(0, [0, 1])
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [])
+
+    def test_delete_many_partial(self):
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [])
+
+        self.set_many(0, {
+            0: {'a': 'b'},
+            1: {'b': 'c'},
+        })
+
+        self.delete_many(0, [0, 1], ['a'])
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [(self._block(1), {'b': 'c'})])
+
+    def test_delete_many_last_field(self):
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [])
+
+        self.set_many(0, {
+            0: {'a': 'b'},
+            1: {'b': 'c'},
+        })
+
+        self.delete_many(0, [0, 1], ['a', 'b'])
+        self.assertItemsEqual(self.get_many(0, [0, 1]), [])
+
```

Comment:
This test is supposed to be testing what happens if you delete the last field from a block (that is, so that there are no fields left).